src/components/CouponDetails/index.jsx

```diff
@@ -381,9 +438,11 @@ class CouponDetails extends React.Component {
           onChange={(checked) => {
             this.handleCodeSelection({ checked, code });
             if (checked) {
-              this.selectedTableRows[code.code] = true;
-            } else if (checked && this.selectedTableRows[code.code]) {
-              delete this.selectedTableRows[code.code];
+              if (this.selectedTableRows[code.code]) {
```

Comment:
[clarification] Is this change changing the underlying logic here?  Before this conditional explicitly only applied when `checked` is truthy; however in the updated line, `checked` is no longer part of the condition. I believe we'd be trying to delete the selected table row even when its not selected/checked as is?

Reply:
This is now coming under `checked if `statement, it will be reach only if `checked` is truthy in parent `if`
```
if (checked) {
  if (this.selectedTableRows[code.code]) {
    delete this.selectedTableRows[code.code];
  } else {
    this.selectedTableRows[code.code] = true;
  }
}
```